app/uk/gov/hmrc/eeittadminfrontend/controllers/GformsController.scala

```diff
@@ -16,10 +16,80 @@
 
 package uk.gov.hmrc.eeittadminfrontend.controllers
 
+import play.api.data.Form
+import play.api.data.Forms._
 import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.Json
+import uk.gov.hmrc.eeittadminfrontend.AppConfig
+import uk.gov.hmrc.eeittadminfrontend.config.Authentication
+import uk.gov.hmrc.eeittadminfrontend.connectors.GformConnector
+import uk.gov.hmrc.eeittadminfrontend.models.{FormTypeId, GformIdAndVersion}
 import uk.gov.hmrc.play.frontend.auth.Actions
 import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
 import uk.gov.hmrc.play.frontend.controller.FrontendController
 
-class GformsController(val authConnector: AuthConnector, val messagesApi: MessagesApi) extends FrontendController with Actions with I18nSupport
+import scala.concurrent.Future
+
+class GformsController(val authConnector: AuthConnector)(implicit appConfig: AppConfig, val messagesApi: MessagesApi) extends FrontendController with Actions with I18nSupport {
+
+  def getGformByFormType = Authentication.async { implicit request =>
+    gFormForm.bindFromRequest().fold(
+      formWithErrors => {
+        Future.successful(BadRequest(uk.gov.hmrc.eeittadminfrontend.views.html.gform_page(gFormForm)))
+      },
+      gformIdAndVersion =>
+        GformConnector.getGformsTemplate(gformIdAndVersion.formTypeId, gformIdAndVersion.version).map { x =>
```

Comment:
fold should be your friend :)

```
 .... map( _.fold(Ok("Error"), x => Ok(Json.prettyPrint(x))))
```
This is easier to read/write than 3 lines of pattern matching.

Now in terms of reporting errors. When _item_ is not found you might want to return `NotFound(...)` rather than `Ok(...)`. This will change the http status code to `404`.